kotsadm/pkg/socketservice/socketservice.go

```diff
@@ -515,9 +531,29 @@ func checkRestoreComplete(a *apptypes.App, restore *velerov1.Restore) error {
 			return errors.New("backup is missing required annotations")
 		}
 
-		sequenceStr, ok := backupAnnotations["kots.io/app-sequence"]
-		if !ok || sequenceStr == "" {
-			return errors.New("backup is missing sequence annotation")
+		sequenceStr := ""
+		if backupAnnotations["kots.io/instance"] == "true" {
+			b, ok := backupAnnotations["kots.io/apps-sequences"]
+			if !ok || b == "" {
+				return errors.New("instance backup is missing apps sequences annotation")
+			}
+
+			var appsSequences map[string]string
+			if err := json.Unmarshal([]byte(b), appsSequences); err != nil {
+				return errors.Wrap(err, "failed to unmarshal apps sequences")
+			}
+
+			s, ok := appsSequences[a.Slug]
+			if !ok || s == "" {
+				return errors.New("instance backup is missing sequence annotation")
+			}
+			sequenceStr = s
+		} else {
+			s, ok := backupAnnotations["kots.io/app-sequence"]
+			if !ok || s == "" {
+				return errors.New("backup is missing sequence annotation")
+			}
+			sequenceStr = s
```

Comment:
`ParseInt` can go inside this `else` block if new annotation is `map[string]int`

Reply:
i had it that way, then changed it, dunno why :). will update